SelectBox grouped Items demo: Update pregrouped dataSource

H-Frederick opened the pull request:
Assigned initial values to editors
Called the dataSource's load() method after the dataSource instance is created.

A refactor for ASP.NET Core & MVC is necessary to split away the dataSource configuration from the controls

H-Frederick committed:
Update pregrouped dataSource

Assigned initial values to editors
Called the dataSource's load() method after the dataSource instance is created.

A refactor for ASP.NET Core & MVC is necessary to split away the dataSource configuration

H-Frederick committed:
Update etalon for SelectBox-GroupedItems.png

Updated etalon

ksercs commented on JSDemos/Demos/SelectBox/GroupedItems/React/App.js:
```diff
@@ -26,6 +26,8 @@ const fromPregroupedData = new DataSource({
   },
 });
 
+fromPregroupedData.load();
```
I think it's better to do in component's constructor

ksercs commented on JSDemos/Demos/SelectBox/GroupedItems/Angular/app/app.component.html:
```diff
@@ -7,6 +7,7 @@
         valueExpr="ID"
         [grouped]="true"
         displayExpr="Name"
+        [value]="1"
```
We should use 2-way binding everywhere.
Value change does not work now.

ksercs commented on MVCDemos/Views/SelectBox/GroupedItems.cshtml:
```diff
@@ -10,18 +27,20 @@
             @(Html.DevExtreme().SelectBox()
-                .DataSource(SampleData.ElectronicsGrouped)
-                .DataSourceOptions(o => o.Map("dataSource_map"))
+                .ID("selectbox-pregroupeddata")
```
Why do we need this ID?

ksercs commented on JSDemos/Demos/SelectBox/GroupedItems/React/App.js:
```diff
@@ -37,7 +39,8 @@ class App extends React.Component {
               grouped={true}
-              displayExpr="Name" />
+              displayExpr="Name"
+              value={1} />
```
We should use uncontrolled mode in React.
It's not possible now to change the value through UI
```suggestion
              defaultValue={1} />
```

H-Frederick committed:
Update react and asp.net

Update React (move dataSource.load() to constructor & use uncontrolled mode)

Update ASP.NET (cleanup IDs in configuration)
